docs/template_plugin/tests/functional/op_reference/lstm_sequence.cpp

```diff
@@ -0,0 +1,498 @@
+                                               std::vector<float>{},
+                                               std::vector<float>{},
+                                               std::vector<std::string>{"sigmoid", "tanh", "tanh"},
+                                               params.clip,
+                                               params.input_forget);
+
+        auto function = std::make_shared<Model>(lstm_sequence->outputs(), ParameterVector{X, H_t, C_t, S_t, W, R, B, P});
+        return function;
+    }
+};
+
+TEST_P(ReferenceLSTMSequenceTest, CompareWithRefs) {
+    Exec();
+}
+
+TEST_P(ReferenceLSTMSequenceV1Test, CompareWithRefs) {
+    Exec();
+}
+
+template <element::Type_t ET>
+std::vector<LSTMSequenceParams> generateParamsRuntimeRef(size_t batchSize,
+                                                         size_t inputSize,
+                                                         size_t hiddenSize,
+                                                         size_t seqLength,
+                                                         float clip,
+                                                         op::RecurrentSequenceDirection lstm_direction) {
+    using T = typename element_type_traits<ET>::value_type;
+
+    size_t numDirections = (lstm_direction == op::RecurrentSequenceDirection::BIDIRECTIONAL) ? 2 : 1;
+
+    Shape XShape = Shape{batchSize, seqLength, inputSize};
+    Shape H_tShape = Shape{batchSize, numDirections, hiddenSize};
+    Shape C_tShape = Shape{batchSize, numDirections, hiddenSize};
+    Shape S_tShape = Shape{batchSize};
+    Shape WShape = Shape{numDirections, 4 * hiddenSize, inputSize};
+    Shape RShape = Shape{numDirections, 4 * hiddenSize, hiddenSize};
+    Shape BShape = Shape{numDirections, 4 * hiddenSize};
+    Shape YShape = Shape{batchSize, numDirections, seqLength, hiddenSize};
+    Shape HoShape = Shape{batchSize, numDirections, hiddenSize};
+    Shape CoShape = Shape{batchSize, numDirections, hiddenSize};
+
+    std::vector<T> X = NGraphFunctions::Utils::generateVector<ET>(shape_size(XShape));
+    std::vector<T> H = NGraphFunctions::Utils::generateVector<ET>(shape_size(H_tShape));
+    std::vector<T> C = NGraphFunctions::Utils::generateVector<ET>(shape_size(C_tShape));
+    std::vector<int64_t> S = NGraphFunctions::Utils::generateVector<element::Type_t::i64>(shape_size(S_tShape), seqLength);
+    std::vector<T> W = NGraphFunctions::Utils::generateVector<ET>(shape_size(WShape));
+    std::vector<T> R = NGraphFunctions::Utils::generateVector<ET>(shape_size(RShape));
+    std::vector<T> B = NGraphFunctions::Utils::generateVector<ET>(shape_size(BShape));
+
+    std::vector<std::string> activations = {"sigmoid", "tanh", "tanh"};
+    std::vector<T> Y(shape_size(YShape));
+    std::vector<T> Ho(shape_size(HoShape));
+    std::vector<T> Co(shape_size(CoShape));
+
+    ngraph::runtime::reference::lstm_sequence<T, int64_t>(reinterpret_cast<const char*>(X.data()),
+                   XShape,
```

Comment:
@iefode @ilyachur @itikhono I used runtime reference function of LSTMSequence to get refData because I cannot get reference output data from internal or external(ONNX page) source. Current implement is using random generated input as same as plugin functional test cases.. Now I can make dump of input/output from this function and replace to hard corded values. How do you think about this?

Reply:
It is not a good that you reference implementation to get reference value.
Because in this case this test checks nothing (template plugin also uses reference implementation).

Reply:
@ilyachur Actually, yes, they are using same code to execute, then how about to use hard coded value from the runtime reference code? Otherwise need to consider external framework(ONNX, Tensorflow) to get the data.

Reply:
Why cannot we hardcode some vector of values for these tests?

Reply:
Got it! I will generate the input/refOut data from outside and put them as hard coded as other test cases.